Added a simple Vagrant setup for hacking on FAS3

ryanlerch opened the pull request:
Fixes #208

ryanlerch commented:
@skrzepto thanks! 

I have updated the Vagrantfile to use postgres rather than sqlite.

--ryanlerch

skrzepto commented on Vagrantfile:
```diff
@@ -6,11 +6,18 @@ Vagrant.configure(2) do |config|
   config.vm.box = "f23-cloud-libvirt"
```
Should we bump this to 25 since f23 is eol soon?

ryanlerch commented:
okies, it now uses fedora25 cloud.

ryanlerch commented:
Just updated the Vagrant setup to use ansible as the provisioner, rather than shell scripts. This is how we do it in other places like bodhi.

ryanlerch commented:
Just added liberation-mono-fonts as a requirement for teh Vagrant provisioning process. See #146 for why.

ryanlerch commented:
Just squashed all the previous commits into a single commit on this branch.

puiterwijk commented:
Other than that I hate Vagrant (it doesn't work at all with rpm-ostree setups due to messing around with plugins files and all kinds of other messy stuff), this patch looks fine to me.  :+1: 

Perhaps it's an idea to move Vagrantfile to Vagrantfile.example though, since some people (e.g. me) will need to modify the vagrantfile before they can run it.

laxathom commented:
I think you can remove the `playbook.retry` file.
I'd encourage to have all those file into a dedicated rep though or rename `devel` to something else to better match what is inside (ansible's files).
Other than this, look good to me.

ryanlerch commented:
Okay, 
* Vagrantfile is now renamed to Vagrantfile.example
* added pyramid_debugtoolbar as a dep in the ansible setup to be installed
* Changed the recently deprecated fake-factory with Faker in setup.py
* removed the playbook.retry
* added Vagrantfile, .vagrant & playbook.retry to the gitignore

I kept the ansible files in the devel directory, as this is how is it set up in other repos in Infra. I was using the bodhi Vagrant/ansible setup as a guide to try to keep these setup in similar ways.

laxathom commented:
> I kept the ansible files in the devel directory, as this is how is it set up in other repos in Infra. I was using the bodhi Vagrant/ansible setup as a guide to try to keep these setup in similar ways.

Okay, then, why such a name? I find it quite confusing. *devel* is a short for *development* afaik. 95% of the files are development's files. If I follow this logic, I should put everything single development related file into that directory.

ryanlerch commented:
> Okay, then, why such a name? I find it quite confusing. devel is a short for development afaik. 95% of the files are development's files. If I follow this logic, I should put everything single development related file into that directory.

My interpretation of the name is that it is so people don't assume that it is production-ready ansible setup, but instead designed for development. how would you feel about renaming the folder `devel-ansible` ?

laxathom commented on .gitignore:
```diff
@@ -71,3 +71,8 @@ fas.anjuta
+
+# Vagrant
+.vagrant/
+devel/ansible/playbook.retry
```
Since you changed the rep name, you might want to rename that one too.
And update this entry by `devel-ansible/ansible/*.retry`

jeremycline commented:
I recently updated the [Vagrant Fedora wiki](https://fedoraproject.org/wiki/Vagrant) with the Vagrantfile I typically use, if you want to peruse it. The main differences between it and the one included here are:

* It pulls the box from download.fedoraproject.org directly
* The code is synced to the ~/devel directory (that's a personal preference and it doesn't really matter)
* It makes optional use of the vagrant-hostmanager plugin
* It provides a (commented-out) way to create a dnf cache for the box so vagrant destroy && vagrant up is faster
* It explicitly sets the resources (cpu, memory, etc) used for the VM.

Just an FYI 😃. I think this looks fine to get started with.

ryanlerch commented:
Thanks @jeremycline !

Once we get this one merged, I will look into adding these features.

ryanlerch commented:
@laxathom can merge this one now?
